cmd/csi-node-driver-registrar/node_register.go

```diff
@@ -74,6 +75,44 @@ func nodeRegister(
 		klog.Errorf("Registration Server stopped serving: %v", err)
 		os.Exit(1)
 	}
+
+	//Periodically check whether the socket exists
+	go periodicallyCheck(socketPath)
+
 	// If gRPC server is gracefully shutdown, exit
 	os.Exit(0)
 }
+
+func periodicallyCheck(socketPath string) {
+	t := time.NewTicker(3 * time.Second)
+	defer t.Stop()
+
+	for {
+		select {
+		case <-t.C:
+			_, err := os.Stat(socketPath)
+			if err != nil {
+				if !os.IsNotExist(err) {
+					klog.Errorf("failed to stat the socket %s with error: %+v", socketPath, err)
+					os.Exit(1)
+				} else {
+					_, err := os.Create(socketPath)
```

Comment:
Does the registration server need to be updated to listen to this socket?

Reply:
if  the reg-socket had been deleted,  the socket was always used？It's no necessary to update to listen to this socket. Only create it.